tools/build-id.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/python3
+
+import os
+import subprocess
+import urllib.request
+import zipfile
+import argparse
+
+READELF_TOOL_PATH="/toolchains/llvm/prebuilt/darwin-x86_64/bin/x86_64-linux-android-readelf"
+
+FLAVORS = ['base', 'objectServer']
+ARCHS = ['arm64-v8a', 'armeabi-v7a', 'x86', 'x86_64']
+
+def file_exists(path):
+    f = None
+    try:
+        f = open(path)
+        return True
+    except IOError:
+        return False
+    finally:
+        if f:
+            f.close()
+
+def find_ndk():
+    PROPS_FILE=f"{os.getcwd()}/../realm/local.properties"
```

Comment:
I don't think that projects using https://developer.android.com/studio/projects/install-ndk#apply-specific-version to configure the NDK will have the `ndk.dir` set in `local.properties` so you might want to grab it by some other means. As the version is probably not that important to just read the build ids, you could maybe just grab it from the newest version available from `ANDROID_HOME` (`find $ANDROID_HOME/ndk/ -name ".*" -prune -maxdepth 1 -o -print | sort -rV | head -n 1`)

Reply:
Thanks, I have added this way to resolve the ndk path.